server/src/test/java/io/crate/analyze/AlterTableAddColumnAnalyzerTest.java

```diff
@@ -667,4 +671,50 @@ public void test_default_column_varchar_length_of_the_expression_is_not_lost_aft
             is(10)
         );
     }
+
+    @Test
+    public void add_multiple_columns_pkey_indices_referring_to_correct_ref_indices() throws Exception {
+        e = SQLExecutor.builder(clusterService)
+            .addTable("CREATE TABLE tbl (x int)")
+            .build();
+
+        AnalyzedAlterTableAddColumn analyzedAlterTableAddColumn =
+            e.analyze("ALTER TABLE tbl ADD COLUMN o['a']['b'] int primary key, ADD COLUMN o['a']['c'] int primary key");
+
+        PlannerContext plannerContext = e.getPlannerContext(clusterService.state());
+
+        var request = AlterTableAddColumnPlan.createRequest(
+            analyzedAlterTableAddColumn,
+            plannerContext.nodeContext(),
+            plannerContext,
+            Row.EMPTY,
+            SubQueryResults.EMPTY,
+            e.fulltextAnalyzerResolver()
+        );
+
+
+        assertThat(request.pKeyIndices().size()).isEqualTo(2);
+        assertThat(request.references().size()).isEqualTo(4); // 2 leaves (b, c) and their common parents (o, a).
+
+        List<String> pKeyColNames = new ArrayList<>();
+        for (IntCursor cursor: request.pKeyIndices()) {
+            var ref = request.references().get(cursor.value);
+            pKeyColNames.add(ref.column().leafName());
+        }
+        assertThat(pKeyColNames).containsExactlyInAnyOrder("b", "c");
+    }
+
+    @Test
+    public void add_multiple_columns_adding_same_name_primitive_throws_an_exception() throws IOException {
+        e = SQLExecutor.builder(clusterService)
+            .addTable("CREATE TABLE tbl (x int)")
+            .build();
+
+        assertThrowsMatches(
+            () -> e.analyze("ALTER TABLE tbl ADD COLUMN o['a']['b'] int primary key, ADD COLUMN int_col INTEGER, ADD COLUMN int_col INTEGER"),
```

Comment:
I guess it's the same if the second time the `int_col` has a different type?

Reply:
yep, updated test in https://github.com/crate/crate/pull/13383/commits/595ab4ca6a837601082d3c935f2c38193c932262

Reply:
I would keep both, to be on the safe side.